pkg/target/cluster.go

```diff
@@ -47,6 +52,19 @@ func (cl *Cluster) resolveHosts(path string) ([]*Host, error) {
 		return []*Host{
 			cl.Hosts[jumpHash(path, len(cl.Hosts))],
 		}, nil
+	case conf.LB:
+		cl.Hm.RLock()
+		defer cl.Hm.RUnlock()
+		availableHostCount := len(cl.AvailableHosts)
+		key := fnv1a.HashString64(path)
+		if availableHostCount == 0 {
+			return []*Host{
+				cl.Hosts[key%uint64(len(cl.Hosts))],
+			}, nil
```

Comment:
when this happens, a record is basically throttled, but we don't count it. we should probably count it to the main queue's `ThrottledRecords`.

Reply:
This is added as a failsafe mechanism.

Reply:
By pushing it on to the host queue, you increase the probability of the records being sent through (network glitches etc) compared to dropping straightway, in which case therecord is definitely dropped.

Reply:
I am not suggesting pushing it to the host queue. We still drop it. I suggest counting the `ThrottledRecords` metric additionally.